asv/benchmark.py

```diff
@@ -316,7 +381,7 @@ def method_caller():
                 code = method_caller.__code__
 
             profile.runctx(
-                code, {'run': self.func}, {}, filename)
+                code, {'run': self.do_run}, {}, filename)
```

Comment:
This will add extra outer stack frames to the profiling output.  In general, we may need to think more carefully about how parameterized benchmarks work with profiling -- one profile per entry in the parameter matrix may make more sense.  But that's a detail for now.  Maybe as a stop gap we use `do_run` here if the benchmark is parameterized, otherwise use `func`.